macro/src/main/scala/SparkSchemaHelper.scala

```diff
@@ -11,7 +11,18 @@ class SparkSchemaHelperImpl(val c: Context) {
     import c.universe._
 
     val constructor = weakTypeOf[T].decl(termNames.CONSTRUCTOR).asMethod
-    val params = constructor.paramLists.flatten.map(param => q"$x.${param.name.toTermName}")
+    val params = constructor.paramLists.flatten
+      .map(param => {
+        val baseExpr = q"$x.${param.name.toTermName}"
+        def toSqlTimestamp(expr: c.Tree) = q"new java.sql.Timestamp($expr.toEpochMilli())"
```

Comment:
Hmm, I found a better way of converting between Timestamp and Instant.
See https://www.baeldung.com/java-time-instant-to-java-sql-timestamp

I think maybe the current solution will loose zone information, while the approach in the link above will keep the zone info,